tests/api-watch-diff/test.ml

```diff
@@ -2,3 +2,179 @@ let%expect_test "test_diff_interface" =
+  let mod_add_signature : signature =
+    [
+      Sig_module
+        ( Ident.create_persistent "M",
+          Mp_present,
+          {
+            md_type =
+              Mty_signature
+                [
+                  Sig_value
+                    ( Ident.create_local "x",
+                      {
+                        val_type =
+                          Transient_expr.type_expr
+                            (Transient_expr.create
+                               (Tconstr
+                                  ( Path.Pident (Ident.create_persistent "int"),
+                                    [],
+                                    ref Mnil ))
+                               ~level:0 ~scope:0 ~id:0);
+                        val_kind = Val_reg;
+                        val_attributes = [];
+                        val_loc = Location.none;
+                        val_uid = Uid.internal_not_actually_unique;
+                      },
+                      Exported );
+                ];
+            md_attributes = [];
+            md_loc = Location.none;
+            md_uid = Uid.internal_not_actually_unique;
+          },
+          Trec_not,
+          Exported );
+      Sig_module
+        ( Ident.create_persistent "N",
+          Mp_present,
+          {
+            md_type =
+              Mty_signature
+                [
+                  Sig_value
+                    ( Ident.create_local "y",
+                      {
+                        val_type =
+                          Transient_expr.type_expr
+                            (Transient_expr.create
+                               (Tconstr
+                                  ( Path.Pident (Ident.create_persistent "float"),
+                                    [],
+                                    ref Mnil ))
+                               ~level:0 ~scope:0 ~id:0);
+                        val_kind = Val_reg;
+                        val_attributes = [];
+                        val_loc = Location.none;
+                        val_uid = Uid.internal_not_actually_unique;
+                      },
+                      Exported );
+                ];
+            md_attributes = [];
+            md_loc = Location.none;
+            md_uid = Uid.internal_not_actually_unique;
+          },
+          Trec_not,
+          Exported );
+    ]
+  in
+  let result =
+    Api_watch_diff.diff_interface ~reference:mod_ref_signature
+      ~current:mod_add_signature
+  in
+  Format.printf "%b" result;
+  [%expect {|true|}]
+
+(* Test Api_watch_diff.diff_interface with the interface mod_remove_signature *)
+let%expect_test "Remove a module from the interface" =
+  let mod_remove_signature : signature = [] in
+  let result =
+    Api_watch_diff.diff_interface ~reference:mod_ref_signature
+      ~current:mod_remove_signature
+  in
+  Format.printf "%b" result;
+  [%expect {|true|}]
+
+(* Test Api_watch_diff.diff_interface with the interface mod_modify_signature *)
```

Comment:
I think this comment doesn't clarify anything that isn't already apparent in the test declaration. What do you think?

Reply:
It sure does not. It's a habit of documenting code but I admit sometimes it can be quite too much. I will try to tone it down a bit. Thank you for the input @shonfeder .